_pages/mobile-interview-project.md

```diff
@@ -8,40 +8,55 @@ header:
+Build an employee directory app that shows a list of employees from the provided endpoint.
+
+The app should display a list (or any kind of collection view!) which shows all the employees returned from the JSON endpoint described below. Each item in the view should contain a summary of the employee, including their photo, name, and team at minimum. You may add more information to the summary if you want, or sort employees in any fashion you’d like – sort and group by name, team, etc.
 
-All screens and views which load from the network should display proper loading, empty, and error states when content is not available. If images fail to load, displaying a placeholder is fine.
+There should be some UX to reload the employee list from within the app at any time. The UX can be done in any way you want: a button, pull-to-refresh, etc.
 
-The employee list ***does not*** need to be persisted to disk – you can reload it from the network on each app launch. On the contrary, images ***should*** be cached on disk as to not waste device bandwidth.
+If there is any additional UI/UX you would like to add, feel free to do so! We only ask that you please do not build any more screens than this list. Do not worry about building custom controls or UI elements – using system-provided, standard elements is totally fine.
 
-You should architect the app in such a way that you can provide unit tests for critical pieces of functionality. There is no minimum or maximum amount of test coverage we’re looking for – instead try to provide enough coverage such that regressions are caught by automated tests. Only worry about unit tests – you can skip view snapshot or integration tests.
+Be sure to appropriately handle the normal variety of errors when querying an endpoint. The app should display useful loading, empty, and error states where appropriate. If images fail to load, displaying a placeholder is fine.
 
-Do not worry about building custom controls and UI elements – using system-provided, standard elements is fine.
+One extra thing we ask is that you please ensure you do not use more network bandwidth than necessary – load expensive resources such as photos on-demand only.
 
-Finally, you should also utilize modern engineering practices, language paradigms, and language features relevant to your mobile platform.
+The employee list should not be persisted to disk. You can reload it from the network on each app launch and when refresh is clicked — but no more often than that unintentionally. (Android developers in particular should take care not to make redundant network calls when the phone is rotated, or when memory is low).
+
+On the contrary, images should be cached on disk so as to not waste device bandwidth (please use an open source image caching solution, or write your own caching – do not rely on HTTP caching for image caching).
```

Comment:
Take out "On the contrary"?

Reply:
Hmm. This is tricky. I think Picasso actually uses HTTP caching for image caching?

Reply:
Maybe leave it in. I'm waffling. Don't mind me.

Reply:
It seems a little grandiloquent to me. It's an instruction we're providing - we can reword to be more easily understood as an instruction. How about just: 

Images should be cached on disk to not waste device bandwidth.
You may use either an open source image caching solution, or write your own caching.
Do not rely on HTTP caching for image caching.

Reply:
Images _should_ be cached on disk to prevent wasting device bandwidth. You may use an open source image caching solution, or write your own caching. 
Do not rely upon HTTP caching for image caching.